src/smt/smt_engine.cpp

```diff
@@ -4284,6 +4285,13 @@ void SmtEnginePrivate::processAssertions() {
   Trace("smt-proc") << "SmtEnginePrivate::processAssertions() : post-simplify" << endl;
   dumpAssertions("post-simplify", d_assertions);
 
+  if (options::symmetryDetect())
+  {
+    SymmetryDetect symd;
+    vector<vector<Node>> part;
+    symd.getPartition(part, d_assertions.ref());
```

Comment:
It is not clear to me this is the right home for invoking this as a pre-processing phase. Who is in charge of refactoring/cleaning up the preprocessing passes? It may make sense to add them as a reviewer for the changes to the SMTEngine.

Reply:
I agree that this should go to preprocessing.
Notice that this is an intermediate commit (SymmetryDetect is a general utility that eventually will be used by a preprocessing pass).

Reply:
@PaulMeng, can you move these files to src/preprocessing/ ?

Reply:
Done